library/src/main/java/com/amar/library/ui/StickyScrollView.kt

```diff
@@ -0,0 +1,161 @@
+package com.amar.library.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.os.Parcelable
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewTreeObserver.OnGlobalLayoutListener
+import androidx.core.widget.NestedScrollView
+import com.amar.library.R
+import com.amar.library.provider.ResourceProvider
+import com.amar.library.provider.ScreenInfoProvider
+import com.amar.library.provider.interfaces.IResourceProvider
+import com.amar.library.provider.interfaces.IScreenInfoProvider
+import com.amar.library.ui.interfaces.IScrollViewListener
+import com.amar.library.ui.presentation.IStickyScrollPresentation
+import com.amar.library.ui.presenter.StickyScrollPresenter
+
+class StickyScrollView : NestedScrollView, IStickyScrollPresentation {
+
+    private var scrollViewListener: IScrollViewListener? = null
+    private var stickyFooterView: View? = null
+    private var stickyHeaderView: View? = null
+    private lateinit var mStickyScrollPresenter: StickyScrollPresenter
+
+
+    constructor(context: Context) : this(context, null) {
+        initLayout()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0) {
```

Comment:
This can be moved to the class definition itself with @JvmOverloads. init {} block can be used for below statements.